streaming/base/storage/download.py

```diff
@@ -35,25 +37,202 @@
 """
 
 
-def download_from_s3(remote: str, local: str, timeout: float) -> None:
-    """Download a file from remote AWS S3 (or any S3 compatible object store) to local.
+class CloudDownloader(abc.ABC):
+    """Download files from a remote storage to a local filesystem."""
 
-    Args:
-        remote (str): Remote path (S3).
-        local (str): Local path (local filesystem).
-        timeout (float): How long to wait for shard to download before raising an exception.
-    """
+    @classmethod
+    def get(cls, remote_dir: Optional[str] = None) -> 'CloudDownloader':
+        """Get the downloader for the remote path.
 
-    def _download_file(unsigned: bool = False,
-                       extra_args: Optional[dict[str, Any]] = None) -> None:
-        """Download the file from AWS S3 bucket. The bucket can be either public or private.
+        Args:
+            remote (str | None): Remote path.
+
+        Returns:
+            CloudDownloader: Downloader for the remote path.
+
+        Raises:
+            ValueError: If the remote path is not supported.
+        """
+        if remote_dir is None:
+            return _DEFAULT_DOWNLOADER()
+
+        prefix = urllib.parse.urlparse(remote_dir).scheme
+        if prefix == 'dbfs' and remote_dir.startswith('dbfs:/Volumes'):
+            prefix = 'dbfs-uc'
+
+        if prefix not in DOWNLOADER_MAPPINGS:
+            raise ValueError(f'Unsupported remote path: {remote_dir}')
+
+        return DOWNLOADER_MAPPINGS[prefix]()
```

Comment:
It's more like every time you call it and get back a single `CloudDownloader` object, it is tied to a single backend yes, but whoever uses it can create as many instances as they want.

Reply:
well...yeah but the .get() function would only allow for one instance per backend. Which is totally fine, just confirming my understanding

Reply:
:+1: